packages/fury-adapter-oas3-parser/lib/parser/oas/parseComponentsObject.js

```diff
@@ -31,9 +31,17 @@ function parseComponentsObject(context, element) {
   const validateIsObject = key => R.unless(isObject,
     createWarning(namespace, `'${name}' '${key}' is not an object`));
 
+  const parseSchemaMember = member => pipeParseResult(namespace,
+    R.compose(parseSchemaObject(context), getValue),
+    (dataStructure) => {
+      // eslint-disable-next-line no-param-reassign
+      dataStructure.content.id = member.key.clone();
```

Comment:
In https://github.com/apiaryio/api-elements.js/pull/86 I moved this logic a bit too far away and into parseOpenAPI and I think instead it should be here inside the components parser so the components contain the ID used for referencing.